### ISSUE TITLE
Compatibility with AWS ES Service

### ISSUE THREAD

**Issue by univerio** [elastic/curator#491]
Curator at the moment is sightly incompatible with the AWS hosted ElasticSearch Service when using the `optimize` command because AWS ES does not expose the `/_cluster/state/metadata` endpoint (returns 401), which curator uses to determine if an index is closed.

I understand that perhaps the solution is to ask AWS to expose the endpoint. I've made a thread about it [here](https://forums.aws.amazon.com/thread.jspa?threadID=217758).

On the curator side, though, perhaps it could warn when it can't determine if a index is closed instead of crashing with an exception. Presumably it would just fail to optimize anyway if it's closed?

**Comment by untergeek** [comment]
@univerio I'm sorry to hear you're having difficulty with AWS ES.  An endpoint-limited version of ES is not a scenario I could have planned for.  Curator was designed to run with full-service versions of Elasticsearch because there was never an expectation there would ever be anything but.

Curator depends on the `index_closed` method in more places than just the `optimize` method, and is also referenced by the `prune_closed` method.  I can't just turn off or ignore exceptions for this check without repercussions.  These methods were added _because_ of the program-crashing results of trying to perform operations on closed indices. The only 2 things you can do to closed indices are open and delete (and delete actually briefly opens the index—internally to Elasticsearch—before deleting).  While I appreciate that AWS ES does not allow users to even close indices, I'm concerned that the path you recommend—making Curator regard the exception as only a warning—could have some unintended consequences for Elasticsearch deployments which can close indices.

If you (or someone else) were to code up a way around this, and add tests to guarantee reverse compatibility with all versions of Elasticsearch from 1.0.x - 2.0.x, I'd be willing to merge that in, if it passes code review.

**Comment by univerio** [comment]
That's unfortunate. What do you think a workaround would look like? A command-line flag that turns on AWS ES-specific behavior? Or trying to automatically detect that the ES instance is from AWS?

**Comment by untergeek** [comment]
I suppose you could parse that out.  I'd think it'd be easier to use the `_cat` API with the JSON output.  The AWS ES version is 1.5+, and the _cat API will show index state (open or closed), but only in Elasticsearch versions > 1.4.

Modifying the `index_closed` function to do an ES version check (Curator does this in several places already), then use the `_cat` API first if an acceptable version of ES is found would solve this in a less fragile way.

**Comment by univerio** [comment]
Great, I'll prepare a pull request.

Side note, I couldn't find a way to make `/_cat/indices/*` output JSON, but using `?h=status` works just as well.

**Comment by untergeek** [comment]
```python
$ python
Python 2.7.10 (default, Aug 22 2015, 20:33:39)
[GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.0.59.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import elasticsearch
>>> client = elasticsearch.Elasticsearch(host='172.19.73.2')
>>> client.cat.indices(format='json',h='index,status')
[{u'status': u'open', u'index': u'logstash-2015.10.14'}, {u'status': u'open', u'index': u'.marvel-2015.10.05'}, {u'status': u'open', u'index': u'logstash-2015.10.10'}, {u'status': u'close', u'index': u'.marvel-2015.10.01'}, {u'status': u'open', u'index': u'.marvel-2015.10.07'}, {u'status': u'open', u'index': u'.marvel-2015.10.06'}, {u'status': u'open', u'index': u'.marvel-2015.10.10'}, {u'status': u'open', u'index': u'.marvel-2015.10.02'}, {u'status': u'open', u'index': u'logstash-2015.10.06'}, {u'status': u'open', u'index': u'test_batch'}, {u'status': u'close', u'index': u'logstash-2015.10.01'}, {u'status': u'open', u'index': u'logstash-2015.10.15'}, {u'status': u'open', u'index': u'.marvel-2015.10.11'}, {u'status': u'open', u'index': u'logstash-2015.10.04'}, {u'status': u'open', u'index': u'.marvel-2015.10.08'}, {u'status': u'open', u'index': u'.marvel-2015.10.15'}, {u'status': u'open', u'index': u'logstash-2015.10.02'}, {u'status': u'open', u'index': u'.marvel-2015.10.13'}, {u'status': u'open', u'index': u'logstash-2015.10.12'}, {u'status': u'open', u'index': u'logstash-2015.10.07'}, {u'status': u'open', u'index': u'logstash-2015.10.08'}, {u'status': u'open', u'index': u'.marvel-2015.10.14'}, {u'status': u'open', u'index': u'.marvel-2015.10.04'}, {u'status': u'open', u'index': u'logstash-2015.10.05'}, {u'status': u'open', u'index': u'.marvel-2015.10.03'}, {u'status': u'open', u'index': u'.marvel-2015.10.12'}, {u'status': u'open', u'index': u'.marvel-kibana'}, {u'status': u'open', u'index': u'.marvel-2015.10.09'}, {u'status': u'open', u'index': u'logstash-2015.10.13'}, {u'status': u'open', u'index': u'.kibana'}, {u'status': u'open', u'index': u'logstash-2015.10.09'}, {u'status': u'open', u'index': u'logstash-2015.10.03'}, {u'status': u'open', u'index': u'logstash-2015.10.11'}]
```

**Comment by ulsa** [comment]
Great, guys. Looking forward to trying this when it appears.

**Comment by untergeek** [comment]
fixed in #499
